UI/GovernmentWnd.cpp

```diff
@@ -0,0 +1,1538 @@
+public:
+    /** \name Structors */ //@{
+    PolicyPalette(GG::X w, GG::Y h);
+    void CompleteConstruction() override;
+    //@}
+
+    /** \name Mutators */ //@{
+    void SizeMove(const GG::Pt& ul, const GG::Pt& lr) override;
+
+    void ShowCategory(const std::string& category, bool refresh_list = true);
+    void ShowAllCategories(bool refresh_list = true);
+    void HideCategory(const std::string& category, bool refresh_list = true);
+    void HideAllCategories(bool refresh_list = true);
+    void ToggleCategory(const std::string& category, bool refresh_list = true);
+    void ToggleAllCategories(bool refresh_list = true);
+
+    void ToggleAvailability(const Availability::Enum type);
+
+    void Populate();
+    //@}
+
+    mutable boost::signals2::signal<void (const Policy*, GG::Flags<GG::ModKey>)> PolicyClickedSignal;
+    mutable boost::signals2::signal<void (const Policy*)> PolicyDoubleClickedSignal;
+    mutable boost::signals2::signal<void (const Policy*, const GG::Pt& pt)> PolicyRightClickedSignal;
+    mutable boost::signals2::signal<void (const std::string&)> ClearPolicySignal;
+
+private:
+    void DoLayout();
+
+    /** A policy type click with ctrl obsoletes policy. */
+    void HandlePolicyClicked(const Policy*, GG::Flags<GG::ModKey>);
+    void HandlePolicyRightClicked(const Policy*, const GG::Pt& pt);
+
+    std::shared_ptr<PoliciesListBox>                        m_policies_list = nullptr;
+    std::map<std::string, std::shared_ptr<CUIStateButton>>  m_category_buttons;
+
+    // Holds the state of the availabilities filter.
+    AvailabilityManager                         m_availabilities_state{false, true, false};
+    std::tuple<std::shared_ptr<CUIStateButton>, // first not used for obsolete for policies
+               std::shared_ptr<CUIStateButton>,
+               std::shared_ptr<CUIStateButton>> m_availabilities_buttons{nullptr, nullptr, nullptr};
+};
+
+GovernmentWnd::PolicyPalette::PolicyPalette(GG::X w, GG::Y h) :
+    GG::Wnd(GG::X0, GG::Y0, w, h, GG::ONTOP | GG::INTERACTIVE)
+{}
+
+void GovernmentWnd::PolicyPalette::CompleteConstruction() {
+    SetChildClippingMode(ClipToClient);
+
+    m_policies_list = GG::Wnd::Create<PoliciesListBox>(m_availabilities_state);
+    AttachChild(m_policies_list);
+    m_policies_list->PolicyClickedSignal.connect(
+        boost::bind(&GovernmentWnd::PolicyPalette::HandlePolicyClicked, this, _1, _2));
+    m_policies_list->PolicyDoubleClickedSignal.connect(
+        PolicyDoubleClickedSignal);
+    m_policies_list->PolicyRightClickedSignal.connect(
+        boost::bind(&GovernmentWnd::PolicyPalette::HandlePolicyRightClicked, this, _1, _2));
+    m_policies_list->ClearPolicySignal.connect(ClearPolicySignal);
+
+    // class buttons
+    for (auto& category : GetPolicyManager().PolicyCategories()) {
+        // are there any policies of this class?
+        bool policy_of_this_class_exists = false;
+        for (const auto& entry : GetPolicyManager()) {
```

Comment:
Use `std::none_of` with a lambda.

```suggestion
    if (std::none_of(GetPolicyManager().begin(), GetPolicyManager().end(), [category](auto& e){ return e.second && category == e.second->Category(); }))
       continue;
```